src/xrpld/app/main/GRPCServer.cpp

```diff
@@ -381,8 +278,7 @@ GRPCServerImpl::CallData<Request, Response>::clientIsUnlimited()
     if (!getUser())
         return false;
     auto clientIp = getClientIpAddress();
-    auto proxiedIp = getProxiedClientIpAddress();
```

Comment:
Are we sure this is only used when in reporting mode? This wont break Clio's unlimited use of gRPC, right?

Reply:
I can't prove it, but I think it is. I don't think there is any other gRPC client in practice. This will return `true` for all clients it did before, plus some more. In fact, this would have returned `false` for a reporting mode client acting on behalf of another client. If Clio is unlimited, that must be because it is not configured as a proxy, right?

Reply:
Clio needs its IP in the `secure_gateway` of the `[port_grpc]`. I'm not sure if that makes it a proxy or what but without it we get `tooBusy` sometimes. Keep in mind that Clio uses the `X-User` header to get unlimited role. I'm not sure how any of that relates to this but from your explanation it sounds like some separate thing.

Reply:
Clio is the client.

With reporting mode, the configuration looks like this:

non-reporting rippled <--> reporting rippled <--> app

In this scenario, when the app sends a request to the reporting server, and it doesn't have that information, it forwards the request to the non-reporting server. In that forwarded request, the reporting server is the client, and the app is the proxied client.

If Clio is not attaching the correct field (`client_ip`) to indicate it is forwarding a request on behalf of a proxied client, then it will never be throttled as long as its address (the client address) is in the list of secure gateways.

(Yes, this is confusing. Clio is the "client" in this code block, but the message's `client_ip` field names the "proxied client".)

Reply:
For forwarding to `rippled` what we do with the real client's IP is 
```cpp
if (forwardToRippledClientIp) {
    connectionBuilder.addHeader(
        {boost::beast::http::field::forwarded, fmt::format("for={}", *forwardToRippledClientIp)}
    );
}
```
and we also set the `X-User` to either `clio_admin` or `clio_user` depending whether the connection is an `admin` one (in Clio terms).
However this has nothing to do with gRPC.

For the gRPC connection we don't set the `client_ip` field but we do set the `user` to `ETL` (probably does nothing).
What happens if you don't add Clio's IP to the list of secure gateways, Clio itself gets throttled. Makes sense.
Note that Clio is not doing any gRPC forwarding for Clio's clients, gRPC is only used by Clio to get the data for Clio's use.